designs/SPEC_TEMPLATE.md

```diff
@@ -0,0 +1,49 @@
+
+# Spec Name
+
+* Status: In Review / Reviewed / Implementing / Implemented
+* Author(s): [NuGet](https://github.com/NuGet)
+* Issue: [1](https://github.com/NuGet/Home/issues/1) Issue Title
+* Type: Feature
+
+## Problem Background
+
+Present the background and the motivation for this design. 
+Present evidence why this is an important problem to tackle
+
+## Who are the customers
+
+Customers affected by this change.
+
+## Requirements
+
+Clarify the requirements in a few bullet points
+
+## Goals
+
+Specifically call out the goals for this proposed design.

```

Comment:
Aren't goals and requirements just synonyms for the same thing?  How are the two sections different?

Reply:
You don't have to "satisfy" all requirements.

Reply:
Isn't it the other way around? Using your favourite search engine and asking for "define requirement", I get:

> a thing that is compulsory; a necessary condition

Reply:
Yeah, the other way around. 
Goal is a broader statement. 
Requirements is very specific.

Reply:
Anyway, is there a better way we can represent this? For example, just one goals section, split into mandatory and optional lists?

Reply:
I've seen people use both so frequently, which is why I've left them in. 

I'd appreciate input from others on this.

Reply:
@nuget/nuget-client @karann-msft, @chgill-MSFT I'd appreciate input from other people on this topic.

Reply:
My thought right now is to just remove the "requirements" section. 

Goals is broader and includes more. The problem background can also include the reqs as necessary depending on the author's preference.

Reply:
> Requirements to me sounds like it should be "Constraints". What is unique to this work that you are having to avoid?

I don't really want this template to be "shocking" to people. 
It's a combination of the 4-5 specs I've seen used by team members and I've not seen constraints used. 

I'll just use Goals as I mentioned. 

It's a barebones guide, it doesn't have to capture everything.

Reply:
It doesn't have to capture everything, but why does it have to capture the same thing twice?
I'd still like more explicit difference here if these 2 sections are necessary.

Reply:
> I'd still like more explicit difference here if these 2 sections are necessary.

Feedback received. I removed the `Requirements` section before I re-requested review.

See: https://github.com/NuGet/Home/pull/9312#issuecomment-604036363